src/plugins/gpunetio/gpunetio_backend_aux.h

```diff
@@ -0,0 +1,230 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GPUNETIO_BACKEND_AUX_H
+#define GPUNETIO_BACKEND_AUX_H
+
+#include <atomic>
+#include <cstring>
+#include <iostream>
+#include <mutex>
+#include <sys/types.h>
+#include <thread>
+#include <vector>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#include <doca_buf_array.h>
+#include <doca_ctx.h>
+#include <doca_dev.h>
+#include <doca_error.h>
+#include <doca_gpunetio.h>
+#include <doca_log.h>
+#include <doca_mmap.h>
+#include <doca_rdma.h>
+#include <doca_rdma_bridge.h>
+
+#include "backend/backend_engine.h"
+#include "common/str_tools.h"
+#include "nixl.h"
+
+// Local includes
+#include "common/list_elem.h"
+#include "common/nixl_time.h"
+
+#define DOCA_MAX_COMPLETION_INFLIGHT 128
+#define DOCA_MAX_COMPLETION_INFLIGHT_MASK (DOCA_MAX_COMPLETION_INFLIGHT - 1)
+#define DOCA_DEVINFO_IBDEV_NAME_SIZE 64
+#define RDMA_SEND_QUEUE_SIZE 2048
+#define RDMA_RECV_QUEUE_SIZE (RDMA_SEND_QUEUE_SIZE * 2)
+#define DOCA_POST_STREAM_NUM 4
+#define DOCA_XFER_REQ_SIZE 512
+#define DOCA_XFER_REQ_MAX 32
+#define DOCA_XFER_REQ_MASK (DOCA_XFER_REQ_MAX - 1)
+#define DOCA_ENG_MAX_CONN 20
+#define DOCA_RDMA_CM_LOCAL_PORT_CLIENT 6543
+#define DOCA_RDMA_CM_LOCAL_PORT_SERVER 6544
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+#define DOCA_RDMA_SERVER_ADDR_LEN \
+    (MAX (MAX (DOCA_DEVINFO_IPV4_ADDR_SIZE, DOCA_DEVINFO_IPV6_ADDR_SIZE), DOCA_GID_BYTE_LENGTH))
+#define DOCA_RDMA_SERVER_CONN_DELAY 500 // 500us
+// Pre-fill the whole recv queue with notif once
+#define DOCA_MAX_NOTIF_INFLIGHT RDMA_RECV_QUEUE_SIZE
+#define DOCA_MAX_NOTIF_MESSAGE_SIZE 8192
+#define DOCA_NOTIF_NULL 0xFFFFFFFF
+#define DOCA_MSG_TAG 0xFF
```

Comment:
Please move defines that used only in a .cpp file to this file. And consider using constexpr instead of define.